### PR TITLE
TINY-10436: Remove `force_hex_color` option and enforce colors to always be HEX

### PR DESCRIPTION
**Pull request by danoaky-tiny**
Related Ticket: TINY-10436

Description of Changes:
- Removed `force_hex_color` config option and HEX colours are now always enforced, as if `force_hex_color` was set to `'always'`. They're also now made lower case, as was the case in TinyMCE v5, i.e `#ff0000` instead of `#FF0000`.
- Swapped a variety of tests back to expecting HEX instead of RGB

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [X] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [X] Milestone set
* [x] Docs ticket created (if applicable)

### CODE REVIEW

File: modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts [hunk 1]
```diff
@@ -33,18 +33,25 @@ describe('browser.tinymce.core.html.StylesTest', () => {
...
     );
     assertStyles(styles,
       '   FONT-SIZE  :  10px  ;   COLOR  :  RED   ',
       'font-size: 10px; color: red;'
     );
+    assertStyles(
+      styles,
+      'border: 1px solid rgb(255, 0, 0);',
+      'border: 1px solid rgb(255, 0, 0);'
+      // TODO: color in border style should be in HEX format once https://ephocks.atlassian.net/browse/TINY-9817 is fixed.
```
Comment (shanmen-tiny) [inline modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts:52]: Ticket number doesn't seem to be correct 🤔

**Review by shanmen-tiny — APPROVED** [review]

File: .changes/unreleased/tinymce-TINY-10436-2024-02-27.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+project: tinymce
+kind: Removed
+body: Removed `force_hex_color` option, with the default now being all colors are forced
```
Comment (spocke) [inline .changes/unreleased/tinymce-TINY-10436-2024-02-27.yaml:3]: We never released this feature so this changelog item should just mention what the new behavior is vs the previous release. The `force_hex_color` option was never something that the users saw.

File: modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts [hunk 3]
```diff
@@ -235,35 +242,12 @@ describe('browser.tinymce.core.html.StylesTest', () => {
...
-    }
-  });
-
-  it('TINY-9819: force_hex_color set to "always"', () => {
-    const styles = Styles({ force_hex_color: 'always' });
     assertStyles(styles, 'color: #aabbcc;', 'color: #aabbcc;');
     assertStyles(styles, 'color: rgb(1, 2, 3);', 'color: #010203;');
     assertStyles(styles, 'color: rgba(1, 2, 3, 1);', 'color: #010203;');
     assertStyles(styles, 'color: rgba(1, 2, 3, 0);', 'color: #010203;');
     assertStyles(styles, 'color: rgba(1, 2, 3, 0.5);', 'color: #010203;');
```
Comment (spocke) [inline modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts:251]: So we are just throwing away the alpha channel? Is this really correct. Should it be hex rgba or skipped if it has a alpha? So loading up a document now with an alpha channel on colors will change them all to new colors when you save so data is lost seems a bit radical. This is how it worked in v5: https://fiddle.tiny.cloud/8RqRc8d4X1/0

The rgba is ignored the rgb is converted. With this PR these two colors become the same color.

- [ ] Consider skipping alpha channel or at least do RGBA hex

**Review by spocke — CHANGES_REQUESTED** [review]

File: modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts (hunk 1 above)
Comment (danoaky-tiny) [reply modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts]: Whoops, typo

File: modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts (hunk 3 above)
Comment (danoaky-tiny) [reply modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts:251]: So how about this (similar to v5)?
```ts
// Convert RGB colors to HEX
RgbaColour.fromString(value)
  .filter((rgba) => rgba.alpha === 1)
  .each((rgba) => {
    value = Transformations.rgbaToHexString(RgbaColour.toString(rgba)).toLowerCase();
  });
```
```ts
const styles = Styles();
assertStyles(styles, 'color: #aabbcc;', 'color: #aabbcc;');
assertStyles(styles, 'color: #aabbcc88', 'color: #aabbcc88;');
assertStyles(styles, 'color: #aabbccff', 'color: #aabbccff;');
assertStyles(styles, 'color: rgb(1, 2, 3);', 'color: #010203;');
assertStyles(styles, 'color: rgba(1, 2, 3, 0);', 'color: rgba(1, 2, 3, 0);');
assertStyles(styles, 'color: rgba(1, 2, 3, 0.5);', 'color: rgba(1, 2, 3, 0.5);');
assertStyles(styles, 'color: rgba(1, 2, 3, 1);', 'color: #010203;');
// Exactly what v5 would do:
// assertStyles(styles, 'color: rgba(1, 2, 3, 1);', 'color: rgba(1, 2, 3, 1);');
```

Or do RGBA Hex, which I guess would need an update to the relevant function in Acid?

**Review by ArvinJ-H — APPROVED** [review]

**Review by MitchC1999 — APPROVED** [review]

File: modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts (hunk 3 above)
Comment (TheSpyder) [reply modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts:251]: Make it behave like v5. That was my intent with this change all along!

File: .changes/unreleased/tinymce-TINY-10436-2024-02-27.yaml (hunk 2 above)
Comment (TheSpyder) [reply .changes/unreleased/tinymce-TINY-10436-2024-02-27.yaml:3]: we did, it was added late in the 6.x cycle once a customer complained about the regression
https://www.tiny.cloud/docs/tinymce/latest/content-formatting/#force_hex_color

**Review by TheSpyder — APPROVED** [review]

File: modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts (hunk 3 above)
Comment (danoaky-tiny) [reply modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts]: It should now behave exactly like it did in v5

**Review by spocke — APPROVED** [review]

**Pull request merged**